File: WaterManagerUI/Model/Item/Coltivazione.cs
using Newtonsoft.Json;

namespace WaterManagerUI.Model.Item;

public class Coltivazione
{
    public int id { get; set; }
    public String? raccolto { get; set; }
    public String? irrigazione { get; set; }
    public String? esigenza { get; set; }
    public double? temperatura { get; set; }
    public double? umidita { get; set; }
    public String semina { get; set; }
    public int idCampo { get; set; }

    public Coltivazione()
    {
    }

    public Coltivazione(String raccolto, String irrigazione, String esigenza, Double temperatura,
        Double umidita, int idCampo)
    {
        this.id = 0;
        this.raccolto = raccolto;
        this.irrigazione = irrigazione;
        this.esigenza = esigenza;
        this.temperatura = temperatura;
        this.umidita = umidita;
        this.semina = "";
        this.idCampo = idCampo;
    }

    [JsonConstructor]
    public Coltivazione(int id, string raccolto, string irrigazione, string esigenza, Double temperatura,
        Double umidita, string semina, int idCampo)
    {
        this.id = id;
        this.raccolto = raccolto;
        this.irrigazione = irrigazione;
        this.esigenza = esigenza;
        this.temperatura = temperatura;
        this.umidita = umidita;
        this.semina = semina;
        this.idCampo = idCampo;
    }
    
    public DateTime SeminaAsDateTime()
    {
        DateTime seminaDate;
        if (DateTime.TryParse(semina, out seminaDate))
        {
            return seminaDate;
        }
        else
        {
            return DateTime.MinValue;
        }
    }
}

File: WaterManager/src/main/java/pissir/watermanager/model/item/Coltivazione.java
package pissir.watermanager.model.item;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author Almasio Luca
 * @author Borova Dritan
 * @author Gattico Alessandro
 */
@Setter
@Getter
@NoArgsConstructor
public class Coltivazione {
	
	private int id;
	private String raccolto;
	private String irrigazione;
	private String esigenza;
	private Double temperatura;
	private Double umidita;
	private String semina;
	private int idCampo;
	
	
	public Coltivazione (int id, String raccolto, String irrigazione, String esigenza, Double temperatura,
						 Double umidita, String semina, int idCampo) {
		this.id = id;
		this.raccolto = raccolto;
		this.irrigazione = irrigazione;
		this.esigenza = esigenza;
		this.temperatura = temperatura;
		this.umidita = umidita;
		this.semina = semina;
		this.idCampo = idCampo;
	}
	
}
